app/assets/javascripts/discourse/app/lib/plugin-api.gjs

```diff
@@ -675,6 +675,30 @@ class PluginApi {
       .addButton(callback);
   }
 
+  /**
+   * Add a new button in the topic admin menu.
+   *
+   * Example:
+   *
+   * ```
+   * api.addTopicAdminMenuButton((topic) => {
+   *   return {
+   *     action: () => {
+   *       alert('You clicked on the coffee button!');
+   *     },
+   *     icon: 'coffee',
+   *     className: 'hot-coffee',
+   *     label: 'coffee.title',
+   *   };
+   * });
+   * ```
+   **/
+  addTopicAdminMenuButton(callback) {
```

Comment:
Nice, similar signature to `addPostAdminMenuButton`. Make sure you include this in the PR description and resulting commit, because it's a new feature for theme/plugin devs. 

Also, `PLUGIN_API_VERSION` needs to be bumped and `docs/CHANGELOG-JAVASCRIPT-PLUGIN-API.md` should be updated.

Reply:
If possible, let's add a test for this new plugin api interface. A basic acceptance test would be fine (register a button, load component, make sure button is rendered).